src/AWS.Deploy.ServerMode.Client/ServerModeSession.cs

```diff
@@ -216,6 +217,28 @@ private Task<bool> IsServerAvailable(CancellationToken cancellationToken)
                 cancellationToken);
         }
 
+        private bool CheckDeployToolPathValidity(string deployToolPath)
+        {
+            deployToolPath = deployToolPath.Trim();
+
+            if (string.IsNullOrEmpty(deployToolPath))
+                return false;
+
+            if (deployToolPath.StartsWith(@"\\"))
+                return false;
+
+            if (deployToolPath.Contains("&"))
+                return false;
+
+            if (Path.GetInvalidPathChars().Where(x => deployToolPath.Contains(x)).Any())
+                return false;
+
+            if (!File.Exists(_deployToolPath))
+                return false;
+
+            return true;
+        }
```

Comment:
This is generic method and should be part of Path class. It can be added as extension.
```
bool Path.IsValid(string? path)
```

Reply:
Path is a static class and cannot add extension methods to. I created a utilities class and put the method there.